web/app/components/comment/comment.jsx

```diff
@@ -244,7 +244,7 @@ export default class Comment extends Component {
 
     if (mods.view === 'preview') {
       return (
-        <div className={b('comment', props, defaultMods)}>
+        <div className={b('comment', props, defaultMods)} role="listitem article" aria-level={mods.level}>
```

Comment:
BTW, why do you think that `listitem` must be used here? Are you sure that it's OK to mix roles here?

Reply:
Oh, I've reread your first comment. As far as I understand, it's the experimental part what you were talking about.

What if we use `article` only for Comment component, `list` for `.root__threads` in Root? And for Thread we can always use `listitem` and `list` only when it has some replies inside. So in this case we will have structure like this one:

```
list (.root__threads)
— listitem list (Thread with Threads inside)
— — article (Comment)
— — listitem (Thread without Threads inside)
— — — article (Comment)
— — listitem (Thread without Threads inside)
— — — article (Comment)
— listitem (Thread without Threads inside)
— — article (Comment)
```

I'm not an expert in a11y, but it sounds OK for me. What do you think?

Reply:
Ah, it's pretty simple.

Here we have Threads inside Thread: https://github.com/umputun/remark/blob/master/web/app/components/thread/thread.jsx#L33-L41

So, you can see that nested Threads generated from `replies` array. You can use `replies.length` as a condition for adding `list` role. For example:

```jsx
const roles = ['listitem'].concat(replies.length ? 'list' : []);
return (
  <div className={b('thread', props)} role={roles}>
```

or even simpler:

```jsx
return (
  <div
    className={b('thread', props)}
    role={['listitem'].concat(replies.length ? 'list' : [])}
  >
```

But I'm not sure should we add `list` if child Threads are collapsed? If we shouldn't then conditional must be like this: `(!collapsed && replies.length) ? 'list' : []`. I don't know what the usual a11y pattern for situation like this is.

Reply:
Thanks! Usually if something is expanded or collapsed, we add `aria-expanded` which is either `true` or `false`. BTW, where are the Expand/Collapse (pseudo-)links located?